calabash/Classes/LPI.mm

```diff
@@ -0,0 +1,122 @@
+
+//
+//  LPI.mm
+//
+//  Created by Karl Krukow on 18/08/11.
+//  Copyright (c) 2011 LessPainful. All rights reserved.
+//
+
+#import "CalabashServer.h"
+#import "LPGitVersionDefines.h"
+
+#ifdef LP_SERVER_ID_KEY_VALUE
+static const char* kLPServerIdentifierKeyValue = LP_SERVER_ID_KEY_VALUE;
+#else
+static const char* kLPServerIdentifierKeyValue = "LP_SERVER_ID_KEY_VALUE is an unknown symbol";
+#endif
+
+int lp_tokenize_str (char *str, char c, char ***arr);
+
+int lp_tokenize_str (char *str, char c, char ***arr) {
+  int count = 1;
+  int token_len = 1;
+  int i = 0;
+  char *p;
+  char *t;
+
+  p = str;
+  while (*p != '\0') {
+    if (*p == c) {
+      count++;
+    }
+    p++;
+  }
+
+  *arr = (char**) malloc(sizeof(char*) * count);
+  if (*arr == NULL) { return -1; }
+
+  p = str;
+  while (*p != '\0') {
+    if (*p == c) {
+      (*arr)[i] = (char*) malloc( sizeof(char) * token_len );
+      if ((*arr)[i] == NULL) { return -1; }
+      token_len = 0;
+      i++;
+    }
+    p++;
+    token_len++;
+  }
+
+  (*arr)[i] = (char*) malloc( sizeof(char) * token_len );
+  if ((*arr)[i] == NULL) {  return -1; }
+
+  i = 0;
+  p = str;
+  t = ((*arr)[i]);
+  while (*p != '\0') {
+    if (*p != c && *p != '\0') {
+      *t = *p;
+      t++;
+    } else {
+      *t = '\0';
+      i++;
+      t = ((*arr)[i]);
+    }
+    p++;
+  }
+
+  return count;
+}
```

Comment:
copy-n-paste C code.

What could go wrong?

@john7doe Can you suggest a better way of tokenizing `LPSERVERID=<SHA>` and extracting `<SHA>`?

Reply:
Why tokenize? how about a symbol with just the sha?

Reply:
It is not enough to look for a SHA in the strings with a regex - multiple matches could be returned.

Are you suggesting something like: `LPSERVERID<SHA>` ?

Reply:
No, something like this:

#define LP_SERVER_ID_VALUE "200ea62274aea0cd35616eb9fcf3832026f7a8ea-dirty"
#define LP_SERVER_ID_KEY_VALUE "LPSERVERID="LP_SERVER_ID_VALUE

in calabash/LPGitVersionDefines.h

and then use LP_SERVER_ID_VALUE when comparing

Reply:
Interesting!